src/main/java/org/hyperledger/fabric/sdk/BlockDeserializer.java

```diff
@@ -81,8 +79,12 @@ EnvelopeDeserializer getData(int index) throws InvalidProtocolBufferException {
     }
 
     byte[] getTransActionsMetaData() {
-
-        return block.getMetadata().getMetadata(Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE).toByteArray();
+        List<ByteString> metadataList = block.getMetadata().getMetadataList();
+        if (metadataList != null && metadataList.size() > Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE) {
+            return block.getMetadata().getMetadata(Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE).toByteArray();
+        } else {
+            return new byte[]{};
```

Comment:
Code that calls this methods assumes that the returned array contains the same number of elements as there are transactions within the block. By returning an empty array you are pushing the problem out to multiple other locations in the code where it will be harder to diagnose any failure.

Reply:
Is it means that throw an exception is the best way to handle the problem?

Reply:
Yes, see my initial comment on this line: https://github.com/hyperledger/fabric-sdk-java/pull/229#discussion_r975176362

Reply:
ok，sir，i fix it